src/main/asciidoc/repository-projections.adoc

```diff
@@ -1,163 +1,276 @@
 
-[source,java]
+[[projections.interfaces.closed]]
+=== Closed projections
+
+A projection interface whose accessor methods all match properties of the target aggregate are considered closed projections.
+
+.A closed projection
+====
+[source, java]
 ----
-interface PersonRepository extends CrudRepository<Person, Long> {
+interface NamesOnly {
 
-  NoAddresses findByFirstName(String firstName);
+  String getFirstname();
+  String getLastname();
 }
 ----
+====
 
-Projections declare a contract between the underlying type and the method signatures related to the exposed properties. Hence it is required to name getter methods according to the property name of the underlying type. If the underlying property is named `firstName`, then the getter method must be named `getFirstName` otherwise Spring Data is not able to look up the source property. This type of projection is also called _closed projection_. Closed projections expose a subset of properties hence they can be used to optimize the query in a way to reduce the selected fields from the data store. The other type is, as you might imagine, an _open projection_. 
+If a closed projection is used, Spring Data modules can even optimize the query execution as we exactly know about all attributes that are needed to back the projection proxy.
+For more details on that, please refer to the module specific part of the reference documentation.
 
-[[projections.remodelling-data]]
-== Remodelling data
+[[projections.interfaces.open]]
+=== Open projections
 
-So far, you have seen how projections can be used to reduce the information that is presented to the user. Projections can be used to adjust the exposed data model. You can add virtual properties to your projection. Look at the following projection interface:
+Accessor methods in projection interfaces can also be used to compute new values by using the `@Value` annotation on it:
 
-.Renaming a property
+[[projections.interfaces.open.simple]]
+.An Open Projection
 ====
-[source,java]
+[source, java]
 ----
-interface RenamedProperty {    <1>
+interface NamesOnly {
 
-  String getFirstName();       <2>
-
-  @Value("#{target.lastName}")
-  String getName();            <3>
+  @Value("#{target.firstname + ' ' + target.lastname}")
+  String getFullName();
+  …
 }
 ----
-This projection has the following details:
+====
 
-<1> A plain Java interface making it declarative.
-<2> Export the `firstName`.
-<3> Export the `name` property. Since this property is virtual it requires `@Value("#{target.lastName}")` to specify the property source.
+The aggregate root backing the projection is available via the `target` variable.
+A projection interface using `@Value` is called open projection, and will cause Spring Data not being able to apply query execution optimizations anymore as the SpEL expression could use any attributes of the aggregate root.
```

Comment:
How about: A projection interface using `@Value` is an open projection. Spring Data can't apply query optimizations anymore as the SpEL expression could use any attributes of the underlying object.

Reply:
I'd lose the "anymore". Don't think it adds anything.